modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/IgniteDynamicEnableIndexingRestoreTest.java

```diff
@@ -0,0 +1,399 @@
+            .setWalMode(WALMode.LOG_ONLY);
+
+        cfg.setDataStorageConfiguration(memCfg);
+        cfg.setConsistentId(gridName);
+        cfg.setSqlSchemas(POI_SCHEMA_NAME, WRONG_SCHEMA_NAME);
+
+        return cfg;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTest() throws Exception {
+        stopAllGrids();
+
+        cleanPersistenceDir();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        stopAllGrids();
+
+        cleanPersistenceDir();
+    }
+
+    /**
+     * @throws Exception if failed.
+     */
+    @Test
+    public void testMergeCacheConfig_StartWithInitialCoordinator() throws Exception {
+        testMergeCacheConfig(0, 1);
+    }
+
+    /**
+     * @throws Exception if failed.
+     */
+    @Test
+    public void testMergeCacheConfig_StartWithInitialSecondNode() throws Exception {
+        testMergeCacheConfig(1, 0);
+    }
+
+    /**
+     * @param firstIdx Index of first starting node after cluster stopping.
+     * @param secondIdx Index of second starting node after cluster stopping.
+     */
+    private void testMergeCacheConfig(int firstIdx, int secondIdx) throws Exception {
+        prepareTestGrid();
+
+        // Check when start from firstIdx node.
+        IgniteEx ig = startGrid(firstIdx);
+
+        startGrid(secondIdx);
+
+        ig.cluster().state(ClusterState.ACTIVE);
+
+        awaitPartitionMapExchange();
+
+        performQueryingIntegrityCheck(ig);
+
+        // Restart and start from the beginning.
+        stopAllGrids();
+
+        ig = startGrids(2);
+
+        ig.cluster().state(ClusterState.ACTIVE);
+
+        awaitPartitionMapExchange();
+
+        performQueryingIntegrityCheck(ig);
+    }
+
+    /**
+     * @throws Exception if failed.
+     */
+    @Test
+    public void testFailJoiningNodeBecauseNeedConfigUpdateOnActiveGrid() throws Exception {
+        prepareTestGrid();
+
+        IgniteEx ig = startGrid(1);
+        ig.cluster().state(ClusterState.ACTIVE);
+
+        try {
+            startGrid(0);
+
+            fail("Node should start with fail");
+        }
+        catch (Exception e) {
+            assertThat(X.cause(e, IgniteSpiException.class).getMessage(), containsString("Failed to join node to the active cluster"));
+        }
+    }
+
+    /**
+     * @throws Exception if failed.
+     */
+    @Test
+    public void testFailJoiningNodeDifferentSchemasOnDynamicIndexes() throws Exception {
+        prepareTestGrid();
+
+        IgniteEx ig = startGrid(1);
+        ig.cluster().state(ClusterState.ACTIVE);
+
+        // Enable indexing with different schema name.
+        createTable(ig.cache(POI_CACHE_NAME), WRONG_SCHEMA_NAME);
+
+        try {
+            startGrid(0);
+
+            fail("Node should start with fail");
+        }
+        catch (Exception e) {
+            assertThat(X.cause(e, IgniteSpiException.class).getMessage(), containsString("Failed to join node to the active cluster"));
```

Comment:
Typo: line too long.

Let's assert here that error message contains correct error description `schema 'DOMAIN' from joining node differs to 'DOMAIN_1'`